src/Symfony/Bundle/FrameworkBundle/CacheWarmer/RouterCacheWarmer.php

```diff
@@ -38,14 +38,22 @@ public function __construct(ContainerInterface $container)
      * Warms up the cache.
      *
      * @param string $cacheDir The cache directory
+     *
+     * @throws \LogicException
      */
     public function warmUp($cacheDir)
     {
         $router = $this->container->get('router');
 
         if ($router instanceof WarmableInterface) {
             $router->warmUp($cacheDir);
+
+            return;
+        } else {
+            @trigger_error(sprintf('Passing a %s without implementing %s is deprecated since 4.1.', RouterInterface::class, WarmableInterface::class), \E_USER_DEPRECATED);
```

Comment:
I do not understand why we care about a deprecation if we immediately throw afterwards anyway.

Reply:
We MUST NOT throw (we cannot do it until 5.0). We should have only the deprecation notice.